src/Calculations.cpp

```diff
@@ -80,100 +188,394 @@ void Calculations::calculateF() {
+                        }
+
+                        #pragma omp critical
+                        {
+                            auto force = pi->getF();
+                            force[0] += newForcei[0];
+                            force[1] += newForcei[1];
+                            force[2] += newForcei[2];
+                            pi->setF(force);
+                        }
+                    }
+                }
+                break;
+
+            case 2: // Parallel execution using OpenMP task parallelism
+                #pragma omp parallel
+                {
+                    #pragma omp single
+                    {
+                        for (size_t i = 0; i < center.size(); ++i) {
+                            #pragma omp task firstprivate(i)
+                            {
+                                Particle* pi = center[i];
+                                std::array<double, 3> newForcei = {0, 0, 0};
+
+                                for (size_t j = 0; j < other.size(); ++j) {
+                                    Particle& pj = other[j];
+                                    std::array<double, 3> f_ij{};
+                                    double e = pi->getEpsilon();
+                                    double s = pi->getSigma();
+
+                                    if (pi->getType() == 0 && pj.getType() == 0) { // is Membrane
+                                        if (!pi->isNeighbour(&pj)) {
+                                            f_ij = decideForceMethod(pi, &pj, e, s);
+                                        }
+                                    } else {
+                                        if (pi->getType() != pj.getType()) {
+                                            for (auto& entry : EAndS) {
+                                                if (entry.isRight(pi->getType(), pj.getType())) {
+                                                    e = entry.getEpsilon();
+                                                    s = entry.getSigma();
+                                                    break;
+                                                }
+                                            }
+                                        }
+                                        f_ij = decideForceMethod(pi, &pj, e, s);
+                                    }
+
+                                    newForcei[0] += f_ij[0];
+                                    newForcei[1] += f_ij[1];
+                                    newForcei[2] += f_ij[2];
+                                }
+
+                                #pragma omp critical
+                                {
+                                    auto force = pi->getF();
+                                    force[0] += newForcei[0];
+                                    force[1] += newForcei[1];
+                                    force[2] += newForcei[2];
+                                    pi->setF(force);
```

Comment:
Hi, I think you never copy the current forces into `old_F`, so you velocity updates are wrong (old_F is always zero as set in the beginning). This might explain why you needed to apply much larger force to pull the membrane up or why you Rayleigh Taylor instability problem doesn't seem to work. 

Please correct me if I missed it somewhere.

Reply:
In the SetF() in Particles we always set the old_f corespondingly. So i think thats not the issue here.

Reply:
ah okay, but then there might be another problem. You change the old_f everytime you apply a force so it doesn't always store the force from last step.

For eg. when you have to apply gravity, LJ force and movement force for membrane in same step, you update old_f after each component, so after the the three forces, f contains total force for this step, but old_f contains gravity and LJ (not the last step force)

Does this seem like a potential error?

Reply:
You are right. We change the F multiple times for each particle in each itteration. That results in false results when using Calculations::calculateV(). Especially when considering we are calculating the Forces at multiple steps in our simulation, like in the force calculation between the particles, in the interaction withe the boundarys or the MembraneStabilization. I think we didn't even considered to think about something like this. I was just using the CalculateV and CalculateX as it is and didnt adjust my code regarding to this big mistake.

Reply:
Ah, no worries, all this is part of the work! I just wanted to understand what went wrong as it would be helpful when I offer the lab in next semesters :) 

I will resolve the conversations.